packages/blade/src/utils/getFocusRingStyles/getFocusRingStyles.web.ts

```diff
@@ -1,15 +1,25 @@
-import type { CSSProperties } from 'styled-components';
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
 import type { Theme } from '../../components/BladeProvider';
 import { castWebType, makeMotionTime } from '~utils';
 
-function getFocusRingStyles(theme: Theme): CSSProperties {
+type GetFocusRingProps = {
+  theme: Theme;
+  negativeOffset?: boolean;
+};
+
+/**
+ * @param props.theme Blade Theme Object
+ * @param props.negativeOffset if set the outline offset will be set to -4px, this is useful
+ * in table component where the outline will get cutoff by the table border
```

Comment:
If this is specific to table, lets add outlineOffset to table only?

Reply:
This can be used in other places also if we face similar focus ring cuttoff issues. 

I would rather keep focus ring styles in a single place than spread it all over the codebase, it would become same as before. 

Also the negativeOffset has to match the outline-width, if we put negativeOffsets in Table styles we will have random `-4px` negative offsets all over the place. 

```
&focus-visible: {
 ...getFocusRingStyles(),
 outlineOffset: `-4px` // Developer has no idea why is this -4px, unless they look inside getFocusRingStyles.
}
```